### DISCUSSION TITLE
Next batch / estimated date / alternatives (OpenTherm?)

### DISCUSSION THREAD

**Discussion by Quarco** [john30/ebusd#685] [General]
First of all, I think this is a great project.
But of course... It's not easy to get your hands of one of these nifty devices!
So, are there any alternatives I could use? for example the eBUS coupler from esera.de (also not available by the way). Or is someone here selling his device? :-)

Regards,
Marco

**Comment by zenlight777** [comment]
I've a Weishaupt WTC-25A heating system and explored alternatives like OpenTherm protocol. Anyone having luck in connecting that instead of the eBUS?

**Reply by zenlight777** [reply]
see https://forum.fhem.de/index.php/topic,61017.msg1186030.html#msg1186030

**Comment by G8GYW** [comment]
I'm waiting for PCBs and components to build this adpater:

[https://gitlab.com/fromeijn/ebuzzz-adapter](url)

I had to order 5 PCBs so if it works I may have some spares ...

Mike

**Reply by lyonanderson** [reply]
Hi there, are you going to be selling them made up?

**Reply by G8GYW** [reply]
Sorry, but no. I need to build my own first and get it working with my Vaillant EcoTec Plus. As I know zilch about ebusd (yet) this may take a while. More later ...

Mike

**Reply by G8GYW** [reply]
OK I built this adapter and it works perfectly with ebusd and my boiler. No pots to adjust!

Mike

![ebuzzz](https://user-images.githubusercontent.com/66752188/203500988-f92bda5c-e98b-4bda-8209-b49c9bd6cd99.jpg)

**Reply by lyonanderson** [reply]
Nice! Can you build me one!?

**Reply by yarwoob** [reply]
I have ordered 5 of these to be made from jclpcb so am hoping it will work.

**Reply by lyonanderson** [reply]
Would you be interested in selling one of them?

**Reply by yarwoob** [reply]
Yes was planning to sell them assuming they work.  Was planning to send one to @john30 as well if he is interested in having one for comment.  I don't really understand anything about how this works but at present this seems the only way to get an adapter of any sort.

**Reply by lyonanderson** [reply]
Great! How much will you be selling for?

**Reply by G8GYW** [reply]
Mine works perfectly and I have enough parts left over to build 2 or 3 more but it will take me a while as I'm quite busy at the moment. Where do you guys live?

**Reply by lyonanderson** [reply]
I'm based in the UK

**Reply by G8GYW** [reply]
> I'm based in the UK
That's easy then. Ģive me a few days to build and test and I'll get back to you.

**Reply by yarwoob** [reply]
I'm in the UK as well.  It's costing $150 to have 5 made and delivered to the UK.  I'll sell them at cost + postage.  Not aiming to make a profit.  I had to change (hopefully like for like) a diode and resistor which were out of stock so I hope I haven't ballsed it up.

**Reply by yarwoob** [reply]
@G8GYW I wouldn't mind one of you as well if it's possible?  I know yours work.

**Reply by G8GYW** [reply]
> @G8GYW I wouldn't mind one of you as well if it's possible? I know yours work.

OK but as I said it takes a while to build them. The quad flat pack is tiny! I substituted the 9.1V energy diode with one from Farnell. Slightly larger package but it fits nicely.

**Reply by G8GYW** [reply]
> @G8GYW I wouldn't mind one of you as well if it's possible? I know yours work.

OK but as I said it takes a while to build them. The quad flat pack is tiny! I substituted the 9.1V energy diode with one from Farnell. Slightly larger package but it fits nicely.

**Reply by G8GYW** [reply]
> > @G8GYW I wouldn't mind one of you as well if it's possible? I know yours work.
> 
> OK but as I said it takes a while to build them. The quad flat pack is tiny! I substituted the 9.1V energy diode with one from Farnell. Slightly larger package but it fits nicely.

Zener diode! Curse the auto correct!

**Reply by lyonanderson** [reply]
@G8GYW great - what's best way to contact you? What price are you selling for?

**Reply by yarwoob** [reply]
No problem at all.  I'm in awe of the fact you can solder stuff that small.

**Reply by G8GYW** [reply]
A lifetime in the industry and now it's a retirement hobby (c:  I used a stencil, solder paste and hot air gun for the UART and hand soldered the rest.

**Reply by G8GYW** [reply]
I have 2 CP2104s left so I can make one board each for @lyonanderson and @yarwoob. How does £20 posted sound? I'll get back to you about contacting as I don't think github supports PMs.

**Reply by lyonanderson** [reply]
Perfect! Thanks @G8GYW

**Reply by yarwoob** [reply]
Yes that sounds great.  Thanks @G8GYW

**Reply by john30** [reply]
guys, I really can't recommend using a design without DC-DC coupler based powering of the eBUS side. We found out years ago that this is a bad idea due to the signal degradation that is happening when doing so. It might be fine for a very little number of participants on the bus (e.g. 2 in total) but I myself have faced weird behaviour of components and also destroyed controller units when taking too much power from the bus. That was just not worth it (more than 200 EUR for a single replacement). So I'd urge you to thoroughly think about it before doing this.

ok, warnings aside: since some of you seem to be in GB, would someone mind acting as a kind of distributor for 3.1 adapters and be willing to do so? This would avoid the long delays your country fellows are facing otherwise. The "job" is not suitable for making money though... anyway, please contact me under ebusd(at)ebusd.eu if you're interested!

**Reply by G8GYW** [reply]
Thanks for the heads up John. I'm only using this to monitor my boiler. If I do add any additional devices I will check the bus with a scope to look for degradation.

Mike

**Reply by G8GYW** [reply]
lyonanderson & yarwoob: contact me on exracal.at.gmail.com.

Mike

**Reply by G8GYW** [reply]
> guys, I really can't recommend using a design without DC-DC coupler based powering of the eBUS side. We found out years ago that this is a bad idea due to the signal degradation that is happening when doing so. It might be fine for a very little number of participants on the bus (e.g. 2 in total) but I myself have faced weird behaviour of components and also destroyed controller units when taking too much power from the bus. That was just not worth it (more than 200 EUR for a single replacement). So I'd urge you to thoroughly think about it before doing this.
> 

John, I have looked at the ebus specification and the maximum number of bus participants is specified as Nmax = 70mA/IΔ, where IΔ is the difference between the low and high idle currents. I measured it as 1.25mA on the ebuzzz adapter. Is that really going to be a problem?

Mike

**Reply by G8GYW** [reply]
> @G8GYW I wouldn't mind one of you as well if it's possible? I know yours work.

yarwoob please email me exracal.at.gmail.com

Mike

**Reply by zarch1972** [reply]
> ok, warnings aside: since some of you seem to be in GB, would someone mind acting as a kind of distributor for 3.1 adapters and be willing to do so? This would avoid the long delays your country fellows are facing otherwise. The "job" is not suitable for making money though... anyway, please contact me under ebusd(at)ebusd.eu if you're interested!

Hi John,

I dropped you an email earlier in the month about being a potential GB helper.
Let me know if you need anything from me.

Cheers, Mick

**Reply by bofruehw** [reply]
> guys, I really can't recommend using a design without DC-DC coupler based powering of the eBUS side. We found out years ago that this is a bad idea due to the signal degradation that is happening when doing so. It might be fine for a very little number of participants on the bus (e.g. 2 in total) but I myself have faced weird behaviour of components and also destroyed controller units when taking too much power from the bus. That was just not worth it (more than 200 EUR for a single replacement). So I'd urge you to thoroughly think about it before doing this.
> 

Hi John,
I agree with you. I tried to set up communication with a simpler solution (eBus to WiFi via the ESP8266 ESP-01), but it is not an easy task. I spent a lot of time and I ended with "something" unreliable without a clue what went wrong. Probably you are right that there is no way to power it from eBus. Maybe in some cases but not soldering parts on 40x80 mm prototype board.

Long story short: I decided to buy an almost out-of-the-box solution and my question is when new batch will be available?

THX for reply
Best regards

**Reply by FoliniC** [reply]
I want to buy a tcp/ip version 3 adapter, I saw that the last batch was closed on july.
Is there an approximate date for the next?
Thanks
Carlo

**Comment by Quarco** [comment]
II did some research myself and now I have entered the path of OpenTherm.

What I need for my zone-heating-project was to enable the heating/flame AND get the most data as well. First solution was to do it was to 'disable' OpenTherm and control the heater in the fashioned on/off way (with relay).
Did the trick but then I had zero data of my heating system.

I have a vaillant EcoTherm plus (OT and E-Bus) and with my original thermostat  (OT Anna/Smile) I could get some data in HA. But there is also an e-buss connector available in my boiler.
In my search for a solution e-bus was the first hit I found that could help me with my zone-heating-project.
But as this hardware is near to impossible to buy / get it within weeks or even months I went OT.

First try was this component (I got mine 2nd hand off a national trading site)
https://diyless.com/product/esp8266-thermostat-shield
But I just couldn't get it to work. Drove me completely crazy. Downloaded library needed for this into HomeAssistant but i just couldn't get it to work. Also it seemed as I looked in the code only a couple of sensors were available with mqtt, but i could be wrong. A I said.. didn't get it to work... Anyone who want's to buy this from me and give it a try... DM me :-)

So as a next step I ordered this OpenTherm Gateway:
https://www.nodo-shop.nl/nl/ons-assortiment/211-opentherm-gateway.html
This seems to do the trick. You can add it in HomeAssistant and then you have 175 sensor-readings under the hood. (have to enable them manualy).
Also the original thermostat can also be connected, but you can override / control the heating in HA.
Got it last week and I still haven't figured out all of the possibilities but for my home-zone-heating-project this seems what i need.
You have to do some soldering after you get the item, but the manual is clear with pictures and it's not too hard.


Hope I could be of help with this info to anyone here :-)

**Reply by zenlight777** [reply]
Thanks @Quarco!

**Reply by zenlight777** [reply]
any idea if this would work with Weishaupt WTC 25 A heating system? or does anyone else having this knows?

**Reply by Quarco** [reply]
https://www.gebruikershandleiding.com/Weishaupt-WTC-25-A-Thermo-Condens/preview-handleiding-783305.html
When I search on "OpenTherm" 0 hits. Ebus => multiple hits.

Don't think this heating system has OT..

**Reply by zenlight777** [reply]
Thank you!

**Comment by fwestenberg** [comment]
@john30 I'm really interested in this adapter as well. I placed a reservation at the 5th of October 2022. Any update about estimated dates? Thanks in advance.

**Comment by bofruehw** [comment]
I made a simple HW for protocol converter based on ESP-01. I did it from at-home available parts on the prototype board. It is based on one of the standard eBus to TTL converter circuit diagrams (https://u.easyeda.com/search?wd=ebus). ESP8266-based E-Bus controller Used in Vaillant boiler monitor project (https://github.com/slavikb/BaiMon). ESP-01 instead ESP8266, without useless three R, G and B diodes (it is completely enclosed), and with 5 to 3.3 V. It has been working very well and stable for more than half a year. Just basics, fit for the purpose, to be as small as possible. I had to power it from an old mobile charger because I couldn't get it to be stable powered by eBus, I did my best, but ESP-01 consumes too much. Maybe it is possible but it is better this way.
